packages/trader/src/Stores/Modules/Trading/Helpers/contract-type.js

```diff
@@ -80,12 +82,18 @@ export const ContractType = (() => {
 
                 available_contract_types[type].config = config;
             });
-
+            const hidden_trade_types = Object.entries(LocalStore.getObject('FeatureFlagsStore')?.data ?? {})
+                .filter(([key, value]) => key.startsWith(TRADE_FEATURE_PREFIX) && !value)
+                .map(([key]) => key.replace(TRADE_FEATURE_PREFIX, ''));
             // cleanup categories
             Object.keys(available_categories).forEach(key => {
-                available_categories[key].categories = available_categories[key].categories?.filter(
-                    item => typeof item === 'object'
-                );
+                available_categories[key].categories = available_categories[key].categories?.filter(item => {
+                    return (
+                        typeof item === 'object' &&
+                        // hide trade types with disabled feature flag:
+                        hidden_trade_types?.every(hidden_type => !item.value.startsWith(hidden_type))
+                    );
+                });
```

Comment:
```suggestion
                available_categories[key].categories = available_categories[key].categories?.filter(item => {
                    return (
                        typeof item === 'object' &&
                        // hide trade types with disabled feature flag:
                        DISABLED_CONTRACTS?.every(hidden_type => !item.value == hidden_type)
                    );
                });
```

Reply:
@yashim-deriv  I wanted to avoid checks for specific ones trade_sharkfin, trade_bla_bla because we might have a lot of new trade types in future, even at the same time, and it might be a hell to go to all this places I changed in this PR and add additional checks for trade_sharkfin for example. Please check my explanation here: https://github.com/binary-com/deriv-app/pull/9816/files#r1329706875